code/game/machinery/atm.dm

```diff
@@ -136,13 +136,17 @@
 
 /obj/machinery/atm/Topic(href, href_list)
 	. = ..()
+	var/mob/user = usr
 	if(. || reject_topic)
 		return
 
 	if(href_list["eject_id"])
 		if(isnull(inserted_card))
 			return TRUE
-		inserted_card.forceMove(drop_location())
+		if(in_range(src, user))
+			user.put_in_hands(inserted_card)
+		else
+			inserted_card.forceMove(drop_location())
```

Comment:
```suggestion
		if(!usr.put_in_hands(inserted_card))
			inserted_card.forceMove(drop_location())
```
Machinery Topic() calls ensure the user is both within range and not incapacitated. `put_in_hands()` can fail, so this needs a fallback.